sdk/rust/nym-sdk/src/mixnet/native_client.rs

```diff
@@ -215,11 +215,12 @@ impl MixnetClient {
     /// client.
     pub async fn disconnect(mut self) {
         if self.forget_me.any() {
-            log::info!("Sending forget me request: {:?}", self.forget_me);
+            log::debug!("Sending forget me request: {:?}", self.forget_me);
             match self.send_forget_me().await {
                 Ok(_) => (),
                 Err(e) => error!("Failed to send forget me request: {}", e),
             };
+            tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
```

Comment:
this seems like a very fragile solution if we depend on extra 2s sleep : / 
I wonder, would it make sense to maybe add some `Arc<Notify>` on the `forget_me` message so we could wait (with some timeout) until it gets sent?

Reply:
You're absolutely right, great call on the notify, will look into it quickly

Reply:
Unfortunatley a no go because of serialization.